src/main/java/net/datafaker/transformations/JsonTransformer.java

```diff
@@ -7,149 +7,185 @@
+            sb.append(": ");
+            if (fields[i] instanceof CompositeField) {
+                sb.append(apply(input, (CompositeField) fields[i], i));
+            } else {
+                value2String(((SimpleField) fields[i]).transform(input), sb);
+            }
+            if (i < fields.length - 1) {
+                sb.append(", ");
+            }
+        }
+        sb.append("}");
+        return sb.toString();
     }
 
-    return data.length() > 1 ? "{" + LINE_SEPARATOR + data + LINE_SEPARATOR + "}" : data.toString();
-  }
-
-  @Override
-  public String generate(Schema<IN, ?> schema, int limit) {
-    StringBuilder sb = new StringBuilder();
-    for (int i = 0; i < limit; i++) {
-      sb.append(apply(null, schema, i));
-      if (i < limit - 1) {
-        sb.append(",").append(LINE_SEPARATOR);
-      }
+    @Override
+    public String generate(FakeSequence<IN> input, Schema<IN, ?> schema) {
+        if (input.isInfinite()) {
+            throw new IllegalArgumentException("The sequence should be finite of size");
+        }
+
+        StringJoiner data = new StringJoiner(LINE_SEPARATOR);
+        for (IN in : input) {
+            data.add(apply(in, schema));
+        }
+
+        char[] delimitedBy = delimitedBy(formatAsArray);
+        return data.length() > 1 ? delimitedBy[0] + LINE_SEPARATOR + data + LINE_SEPARATOR + delimitedBy[1]
+            : data.toString();
     }
-    return limit > 1 ? "{" + LINE_SEPARATOR + sb + LINE_SEPARATOR + "}" : sb.toString();
-  }
-
-  private static void value2String(Object value, StringBuilder sb) {
-    if (value == null) {
-      sb.append("null");
-    } else if (value instanceof Integer
-        || value instanceof Long
-        || value instanceof Short
-        || value instanceof BigInteger
-        || value instanceof Boolean
-        || (value instanceof Double
+
+
+    @Override
+    public String generate(Schema<IN, ?> schema, int limit) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < limit; i++) {
+            sb.append(apply(null, schema, i));
+            if (i < limit - 1) {
+                sb.append(",").append(LINE_SEPARATOR);
+            }
+        }
+        char[] delimitedBy = delimitedBy(this.formatAsArray);
+        return limit > 1 ? delimitedBy[0] + LINE_SEPARATOR + sb + LINE_SEPARATOR + delimitedBy[1] : sb.toString();
+    }
+
+    private static void value2String(Object value, StringBuilder sb) {
+        if (value == null) {
+            sb.append("null");
+        } else if (value instanceof Integer
+            || value instanceof Long
+            || value instanceof Short
+            || value instanceof BigInteger
+            || value instanceof Boolean
+            || (value instanceof Double
             && BigDecimal.valueOf((Double) value).remainder(BigDecimal.ONE).doubleValue() == 0)
-        || (value instanceof BigDecimal
+            || (value instanceof BigDecimal
```

Comment:
Why do we need to touch these lines within this PR?

Reply:
ditto - editorconfig violation/drift - this realigns (pardon the pun).